processing/src/main/java/io/druid/segment/DimensionHandlerUtils.java

```diff
@@ -233,7 +233,8 @@ public static Long convertObjectToLong(Object valObj)
     } else if (valObj instanceof Number) {
       return ((Number) valObj).longValue();
     } else if (valObj instanceof String) {
-      return DimensionHandlerUtils.getExactLongFromDecimalString((String) valObj);
+      Long parsedVal = DimensionHandlerUtils.getExactLongFromDecimalString((String) valObj);
+      return parsedVal == null ? 0L : parsedVal;
```

Comment:
Could you cache 0L in a constant of type `Long`? JLS doesn't guarantee it will be boxed to a single object